src/components/recap-email-and-sms-form.vue

```diff
@@ -0,0 +1,264 @@
+const phoneFilled = computed(() => phoneValue.value?.length ?? 0 > 0)
+const emailAndPhoneFilled = computed(
+  () => emailFilled.value && phoneFilled.value
+)
+
+const simulationId = computed(
+  () => !store.calculs.dirty && store.calculs.resultats?._id
+)
+
+const regexAndPhoneTypeIsValid = () => {
+  const regex = new RegExp("^(?:(?:\\+|00)33|0)\\s*[1-9](?:[\\s.-]*\\d{2}){4}$")
+  return (
+    phoneValue.value &&
+    regex.test(phoneValue.value) &&
+    phoneRef.value?.checkValidity()
+  )
+}
+
+const sendRecap = async (surveyOptin) => {
+  if (emailAndPhoneFilled.value) {
+    sendRecapByEmailAndSms(surveyOptin)
+  } else if (emailFilled.value && !phoneFilled.value) {
+    sendRecapByEmail(surveyOptin)
+    phoneInputErrorMessage.value = false
+    store.setFormRecapPhoneState(undefined)
+  } else if (!emailFilled.value && phoneFilled.value) {
+    sendRecapBySms(surveyOptin)
+    emailInputErrorMessage.value = false
+    store.setFormRecapEmailState(undefined)
+  } else {
+    store.setFormRecapState(undefined)
+    phoneInputErrorMessage.value = true
+    emailInputErrorMessage.value = true
+  }
+}
+
+const inputPhoneIsValid = () => {
+  if (phoneRef.value && !regexAndPhoneTypeIsValid()) {
+    phoneInputErrorMessage.value = true
+    store.setFormRecapPhoneState(undefined)
+    phoneRef.value.focus()
+    StatisticsMixin.methods.sendEventToMatomo(
+      EventCategories.GENERAL,
+      "Invalid phone form",
+      router.currentRoute.value.fullPath
+    )
+    return false
+  }
+  phoneInputErrorMessage.value = false
+  return true
+}
+
+const inputEmailIsValid = () => {
+  if (emailRef.value && !emailRef.value.checkValidity()) {
+    store.setFormRecapEmailState(undefined)
+    emailInputErrorMessage.value = true
+    emailRef.value.focus()
+    StatisticsMixin.methods.sendEventToMatomo(
+      EventCategories.GENERAL,
+      "Invalid email form",
+      router.currentRoute.value.fullPath
+    )
+    return false
+  }
+  emailInputErrorMessage.value = false
+  return true
+}
+
+const postFollowup = async (surveyOptin, email?, phone?) => {
+  const uri = `/api/simulation/${simulationId.value}/followup`
+  const payload = {
+    surveyOptin,
+    phone,
+    email,
+  }
+  await axios.post(uri, payload)
+}
+
+const sendRecapByEmailAndSms = async (surveyOptin) => {
+  try {
+    store.setFormRecapState("waiting")
+    if (!inputEmailIsValid()) {
+      store.setFormRecapState(undefined)
+      return
+    }
+    if (!inputPhoneIsValid()) {
+      store.setFormRecapState(undefined)
+      return
+    }
+    postFollowup(surveyOptin, emailValue.value, phoneValue.value)
+    store.setFormRecapState("ok")
+    phoneValue.value = undefined
+    emailValue.value = undefined
+  } catch (error) {
+    store.setFormRecapState("error")
+  }
+}
+
+const sendRecapBySms = async (surveyOptin) => {
+  try {
+    store.setFormRecapPhoneState("waiting")
+    if (!inputPhoneIsValid()) return
```

Comment:
Ici contrairement à au dessus tu ne fais pas le `store.setFormRecapPhoneState(undefined)` c'est normal ?

Reply:
> Yes ok, pas bloquant, mais ça a plus ça place en config comme c'est static vis à vis du build.

Mais ça du coup c'est du bricolage pour dev et voir/cacher les nouvelles modifs. Donc je sais pas si on le met dans le store en AB testing ou si on ajoute directement l'option des SMS.

Reply:
> Ici contrairement à au dessus tu ne fais pas le store.setFormRecapPhoneState(undefined) c'est normal ?

C'est parce que dans la méthode conditionnelle, il est déjà appelé [ici](https://github.com/betagouv/aides-jeunes/blob/441fa083c599905ff557eff969c8481f748bc048/src/components/recap-email-and-sms-form.vue#L61)

Reply:
Comme on peut pas vraiment l'enlever dans `sendRecapByEmailAndSms` j'opterais peut être pour la mettre dans les trois (quitte à les supprimer de `inputPhoneIsValid` et `inputEmailIsValid`) pour uniformiser au max